### DISCUSSION TITLE
Filter vector layer with integer slider

### DISCUSSION THREAD

**Discussion by krvrd** [opengeos/leafmap#370] [Q&A] [answered]
Hello! I use foliumap module to build streamlit app. I want to filter layer with vector polygons by specific attribute. When I use streamlit for that the page reload every time. 
I read [discussion](https://discuss.streamlit.io/t/ipywidgets-wip/3870/14) about ipywidgets — as I understand ipywidgets are still unavailable for streamlit.

Is there any way to reach smooth filtering?

**Comment by giswqs** [comment, marked as answer]
Try pydeck. It will remove and add layers based on your filter, not refreshing the entire map. 
https://docs.streamlit.io/library/api-reference/charts/st.pydeck_chart

**Reply by krvrd** [reply]
Thank you for the answer! It's very helpful.

Did you mean that I should use [that](https://deck.gl/docs/api-reference/extensions/data-filter-extension) deck extension for filtering? Please share if you have some examples how to use that filter in pydeck. 

Probably there is more convenient ways?

**Reply by giswqs** [reply]
No, the `pydeck_chart` is just for visualizing data. The filter part should be done with streamlit components.
